app/models/ext_management_system.rb

```diff
@@ -175,15 +175,27 @@ def self.create_from_params(params, endpoints, authentications)
 
   def edit_with_params(params, endpoints, authentications)
     tap do |ems|
+      endpoints_changed       = false
+      authentications_changed = false
+
       transaction do
         # Remove endpoints/attributes that are not arriving in the arguments above
-        ems.endpoints.where.not(:role => nil).where.not(:role => endpoints.map { |ep| ep['role'] }).delete_all
-        ems.authentications.where.not(:authtype => nil).where.not(:authtype => authentications.map { |au| au['authtype'] }).delete_all
+        endpoints_to_delete       = ems.endpoints.where.not(:role => nil).where.not(:role => endpoints.map { |ep| ep['role'] })
+        authentications_to_delete = ems.authentications.where.not(:authtype => nil).where.not(:authtype => authentications.map { |au| au['authtype'] })
+
+        endpoints_changed       ||= endpoints_to_delete.delete_all > 0
+        authentications_changed ||= authentications_to_delete.delete_all > 0
 
         ems.assign_attributes(params)
-        ems.endpoints = endpoints.map(&method(:assign_nested_endpoint))
+        ems.endpoints       = endpoints.map(&method(:assign_nested_endpoint))
         ems.authentications = authentications.map(&method(:assign_nested_authentication))
 
+        endpoints_changed       ||= ems.endpoints.any?(&:changed?)
+        authentications_changed ||= ems.authentications.any?(&:changed?)
+
+        after_update_endpoints      if endpoints_changed
+        after_update_authentication if authentications_changed
```

Comment:
At this point, the records are not actually saved in the database - is that ok?

Reply:
Yes that was intentional, originally I had these outside the transaction block but the e.g. `stop_event_monitor_queue_on_change` methods check `default_endpoint.changes.include_any?` so we have to run them before saving.

There is a lot of history to untangle here, primarily that these were `before_save` callbacks so they _had_ to check if the relevant values were changed.  If we move to only call these when we know something changed then we could drop that.

I like this a little better let me see if it is feasible to wrap that into this PR.

Reply:
Okay I've updated this to only call the `after_update_{authentication,endpoint}` callbacks if there is a changed endpoint or auth respectively.